### PR TITLE
[IMP] im_livechat: make LivechatButtonView/localStorageChatbotState declarative

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2892100

### CODE REVIEW

File: addons/im_livechat/static/src/models/livechat_button_view.js [hunk 1]
```diff
@@ -92,6 +92,20 @@ registerModel({
...
         },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeLocalStorageChatbotState() {
+            if (!this.sessionCookie) {
+                return clear();
+            }
+            let data = localStorage.getItem(this.sessionCookieKey);
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/livechat_button_view.js:103]: why let?

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/livechat_button_view.js]: Changes done (now uses `const`)

File: addons/im_livechat/static/src/models/livechat_button_view.js [hunk 2]
```diff
@@ -92,6 +92,20 @@ registerModel({
...
+         */
+        _computeLocalStorageChatbotState() {
+            if (!this.sessionCookie) {
+                return clear();
+            }
+            const data = localStorage.getItem(this.sessionCookieKey);
+            if (!data) {
+                return clear();
+            }
+            return JSON.parse(JSON.parse(data));
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/livechat_button_view.js:107]: why two JSON.parse?

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/livechat_button_view.js:107]: This is a mistake, will fix it now

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/livechat_button_view.js]: Fixed

**Review by oomsveta — APPROVED** [review]

**Pull request closed without merging**